apis/r/R/TileDBGroup.R

```diff
@@ -391,6 +395,7 @@ TileDBGroup <- R6::R6Class(
       # we must open a temporary handle for read, to fill the cache.
       group_handle <- private$.tiledb_group
       if (private$.mode == "WRITE") {
+        stopifnot("FIXME" = is.null(private$.group_open_timestamp))
```

Comment:
Is this supposed to remain?

Reply:
@eddelbuettel I changed the WRITE mode warnings to stops which makes this "FIXME" unreachable (also added comments to this effect).

The python implementation doesn't raise a warning but there are [docstring allusions](https://github.com/single-cell-data/TileDB-SOMA/blob/58f528cb602e840b34696b93958f0731e998e0f2/apis/python/src/tiledbsoma/options/_soma_tiledb_context.py#L56-L58) to some of the pitfalls one can fall into by setting the write timestamp for multiple operations. I was personally disturbed enough by this (#919) to resolve to (personally) avoid the feature from then on. Also I think our main use case is that there should *exist some timestamps* where user will get a consistent snapshot, not that there should *exist no timestamps* at which a user would get an inconsistent snapshot. Reasonable people can and did disagree about all this, of course.

tl;dr I think we're fine leaving this unsupported for now, if we do want to work on it (in the next sprint or perhaps much later) then it wouldn't change the API, so needn't block 1.0rc either.

Reply:
Thanks for the explanation.  Now, without going into specifics of this question, I am generally a bigger fan of "document and explain and lead to _preferred_ interfaces" (as we eg do for open and create factories utilising the available-but-discouraged R6 methods by also requiring the 'handshake') and hence not as much a fan of hard locks such as this one.  I will need to ponder this a little more, and we may want to ascertain we get additional reviews and buy in for the PR.